roles/swift-ring/tasks/main.yml

```diff
@@ -13,3 +13,4 @@
   command: sudo -u swiftops /usr/local/bin/swifttool -i /home/swiftops/.ssh/id_rsa
            bootstrap --config /etc/swift/ring_definition.yml --outdir /etc/swift
   when: not should_boostrap|success
+  notify: restart swift services
```

Comment:
About middleware, my understanding was meta/ only include dependencies for other roles.... Also, I couldn't determine right version of keystonemiddleware to install for source - based install. Closest thing in current code-base is: https://github.com/blueboxgroup/ursula/blob/master/playbooks/neutron-openvswitch-to-ml2-vlan-migration.yml (lines 132-144)